### ISSUE TITLE
root privileges

### ISSUE THREAD

**Issue by bosr** [avanov/ansible-galaxy-pyenv#2]
Hi,

I've just tested your playbook, works nicely, but for this I had to modify the `tasks/main.yml` to add `sudo: yes` in all tasks that need root privileges. If I did not do that and instead set `sudo: yes` as a var when calling your role

    - { role: avanov.pyenv, sudo: yes }

then `pyenv` would be installed inside `/root/.pyenv` because the `sudo: no` variables get overridden in your pyenv installation tasks.

Could you please explain what your workflow is for using your role?

Thanks

PS: I'm new to this, and it might be the case that I did something wrong

**Comment by avanov** [comment]
Hi @bosr 

I've never used the role for installing pyenv outside /home. Usually, I know the exact user who's going to use non-system-wide Python interpreter and I specify him/her in the role variables:

```yaml
- role: avanov.pyenv
  pyenv_path: "/home/avanov/pyenv"
  pyenv_owner: avanov
```

**Comment by bosr** [comment]
Makes sense. 
Thanks!

Romain

**Comment by avanov** [comment]
If you find it inconvenient for your purposes, feel free to reopen the ticket. I just need to know the exact case when it's difficult not to use root privileges.

**Comment by bosr** [comment]
Thanks

**Comment by bosr** [comment]
Hi @avanov 

I have tried to use Ansible 1.9 and I ran into the same root issue when using your pyenv role: `apt`-related tasks complain about not having the right permissions.

Here is how I use it:

    - role: ext-python-pyenv
      sudo: yes # unfortunately mandatory at role top-level
      pyenv_path: "{{ ansible_env.HOME }}/.pyenv"
      pyenv_owner: "{{ pyenv_user }}"

This works fine with Ansible 1.8.x (and maybe earlier versions).

I think it's pretty easy to fix: in `tasks/main.yml`:
- all the `sudo: no` calls for non-root tasks should disappear
- there should be `sudo: yes` for root tasks (calls to the `apt` module).

I think this would have the advantage of getting rid of the `sudo: yes` at role top-level, and work immediately with Ansible 1.9. Furthermore, I've seen this in many (all?) other roles.

Cheers!
